### ISSUE TITLE
An issue with windows being restricted to output boundaries.

### ISSUE THREAD

**Issue by vacuumbeef** [riverwm/river#1004]
Hello! Multiple times I saw that constraining windows within the output is an intentional feature, and on one hand I find it cool and useful, but on other hand in some situations... I may have an argument why windows should be able to go beyond output bounds.

Most of the time it feels very right and good that windows remain within the screen and can't go outside the output borders, I really liked that behavior. But sometimes I make music on my pc, and this is when everything fell apart to the point of being forced to look for another wm.

Here I recorded a video for better understanding. I basically can't do anything in arrangement view while the virtual synth window is opened. The only thing I can do is to close the synth window, because it is also not resizable, but this way the workflow becomes so complex and even dangerous (because what if I will close the main window on accident).

<details>
  <summary>Short video demonstration</summary>

https://github.com/riverwm/river/assets/51813411/df4a33fa-bfcb-43b8-b6b5-be13610b590e

</details>

And this is only one example why constraining the window within the output is not a good idea in the end, in my personal opinion. And there can be more examples probably, I can't be so special.

Or maybe I miss something? Absolutely possible.

**Comment by ifreund** [comment]
I personally would handle this situation by sending the non-resizable synth window to a separate tag.

**Comment by vacuumbeef** [comment]
At that point, it would not much easier than to just close it. 
I mean, I wish I could just move the synthesizer window to the side, do something in underlying window and move it back as it was. Moving it to a separate tag, doing something and then bringing back seems to me as much more work. Also almost every virtual synth I ever used had non-resizable windows. Even if it was resizable, it would take so much work for just being able to do regular stuff.
I guess the problem is not very obvious to those who don't do this sort of stuff on a regular basis. Maybe workflow difference.

**Comment by vide0hanz** [comment]
Consider a dual monitor setup as well. The mouse cursor itself can move between outputs, so naturally one might expect to be able to drag a floating window there as well. The current behavior feels restricting in this context.

**Comment by realcharmer** [comment]
I agree. Dragging floating views outside of the display border and between outputs is so natural and self-evident, I always get surprised by River constraining the view inside the visible area. It is very annoying in my opinion. Sometimes I even need to stretch a view across two or more displays for instance, which is not possible at the moment.
